specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/artifacts.json

```diff
@@ -203,6 +203,52 @@
         }
       }
     },
+    "/linkedservices/{linkedServiceName}/rename": {
+      "put": {
+        "tags": [
+          "linkedServices"
+        ],
+        "operationId": "LinkedService_RenameLinkedService",
+        "x-ms-examples": {
+          "LinkedServices_Rename": {
+            "$ref": "./examples/LinkedServices_Rename.json"
+          }
+        },
+        "description": "Renames a linked service.",
+        "x-ms-long-running-operation": true,
```

Comment:
Properties of the resource that is getting renamed are (thanks @srinuthati for explaining this offline)

1. The user provides the original name at the time of creating the resource.
2. The name is the last segment of the path identifying the resource.
3. Other resources can refer to this path, which means all those references are updated internally as part of rename.

Few gotchas of renaming in general, are: (thanks @johanste)

1. it will leave all existing URLs handed out to clients at one point or another with "dangling" pointers 
2. If someone creates a new resource with the old (now vacated name), clients may inadvertently point to the new resource. (This is generally only a problem in security-sensitive scenarios e.g. key-vault)

@srinuthati please make sure to validate these implications in the context of synapse service.

Johan's proposal is to use POST for rename; will run it by the review board. I'll update the decision here.

Reply:
Yes, this should not be a PUT operation. The semantics of PUT would be to replace (or create) the resource indicating the the path (`/linkedservices/{linkedServiceName}/rename`).

Reply:
Will the service redirect clients that use the old name, or will you respond with 404/Not Found?

Reply:
1. Yes we will change this to POST.
2. While the rename is in progress the clients will be redirected to oldname. but once the rename is completed clients will be responded with 404 if they do get using old name. UI handles display of old vs new name based on rename status.